plugin-server/src/main/ingestion-queues/session-recordings-consumer.ts

```diff
@@ -0,0 +1,138 @@
+                continue
+            }
+
+            let messagePayload: RawEventMessage
+            let event: PipelineEvent
+
+            try {
+                // NOTE: we need to parse the JSON for these events because we
+                // need to add in the team_id to events, as it is possible due
+                // to a drive to remove postgres dependency on the the capture
+                // endpoint we may only have `token`.
+                messagePayload = JSON.parse(message.value.toString())
+                event = JSON.parse(messagePayload.data)
+            } catch (error) {
+                status.warn('⚠️', 'invalid_message', {
+                    reason: 'invalid_json',
+                    error: error,
+                    partition: batch.partition,
+                    offset: message.offset,
+                })
+                await producer.queueMessage({ topic: KAFKA_SESSION_RECORDING_EVENTS_DLQ, messages: [message] })
+                continue
+            }
+
+            status.debug('⬆️', 'processing_session_recording', { uuid: messagePayload.uuid })
+
+            if (!messagePayload.team_id && !event.token) {
+                status.warn('⚠️', 'invalid_message', {
+                    reason: 'no_token',
+                    partition: batch.partition,
+                    offset: message.offset,
+                })
+                await producer.queueMessage({ topic: KAFKA_SESSION_RECORDING_EVENTS_DLQ, messages: [message] })
+                continue
+            }
+
+            const teamId =
+                messagePayload.team_id ?? (event.token ? await teamManager.getTeamIdByToken(event.token) : null)
+
+            if (!teamId) {
+                status.warn('⚠️', 'invalid_message', {
+                    reason: 'team_not_found',
+                    partition: batch.partition,
+                    offset: message.offset,
+                })
+                await producer.queueMessage({ topic: KAFKA_SESSION_RECORDING_EVENTS_DLQ, messages: [message] })
+                continue
+            }
+
+            if (event.event === '$snapshot') {
+                await createSessionRecordingEvent(
+                    messagePayload.uuid,
+                    messagePayload.team_id,
+                    event.distinct_id,
+                    parseEventTimestamp(event as PluginEvent),
+                    event.ip,
+                    event.properties || {},
+                    producer
+                )
+            } else if (event.event === '$performance_event') {
+                await createPerformanceEvent(
+                    messagePayload.uuid,
+                    messagePayload.team_id,
+                    event.distinct_id,
+                    event.properties || {},
+                    event.ip,
+                    parseEventTimestamp(event as PluginEvent),
+                    producer
+                )
+            }
```

Comment:
question: How do we know which event types we can get here? Looking at the function name, `startSessionRecordingEventsConsumer`, I would have expected only `$snapshot`. Wondering if there are any docs on event types I can review.

Reply:
Indeed `$performance_event` being processed here is a little confusing. `$performance_event`s as I understand are very much tied to session recordings so it's not completely out of place here but could be processed separately.

I would like us to get to a point that we have some documentation on the valid values e.g. as an [OpenAPI spec](https://github.com/PostHog/posthog/pull/13729/files). You can deduce from e.g. typings within the JS client library or python types but I think we can do better!

Reply:
Cool, good to know they are not out of place :+1:

I'll dig through the Python types/JS client library, but yeah OpenAPI spec docs would be amazing. There are Python libraries that can export/import OpenAPI spec so there's likely something for JS too. Something for the future :+1: